src/utils/did-helper.ts

```diff
@@ -86,6 +87,69 @@ export const resolveDIDDocumentAuth = async (
   );
 };
 
+function emptyStateDID(did: DID) {
+  const id = DID.idFromDID(did);
+  const didType = buildDIDType(
+    DID.methodFromId(id),
+    DID.blockchainFromId(id),
+    DID.networkIdFromId(id)
+  );
+  const identifier = Id.idGenesisFromIdenState(didType, 0n);
+  const emptyDID = DID.parseFromId(identifier);
+
+  return emptyDID;
+}
+
+export const resolveDidDocumentEip712MessageAndSignature = async (
+  did: DID,
+  resolveURL: string,
+  opts?: {
+    state?: Hash;
+    gist?: Hash;
+  }
+): Promise<IdentityStateUpdate | GlobalStateUpdate> => {
+  let didString = did.string().replace(/:/g, '%3A');
+  // for gist resolve we have to `hide` user did (look into resolver implementation)
+  const isGistRequest = opts?.gist && !opts.state;
+  if (isGistRequest) {
+    didString = emptyStateDID(did).string();
+  }
+  let url = `${resolveURL}/1.0/identifiers/${didString}?signature=EthereumEip712Signature2021`;
```

Comment:
resolverUrl

Reply:
should we add signature as option and rename method to resolve DidDocument with a result DIDResolutionMetadata and then the one who called function can parse signature if persists?